webapps/api/oauth2/redirectview.py

```diff
@@ -67,14 +67,12 @@ def get(self, request, *args, **kwargs):
         base_url = Configuration.get('/ovs/framework/webapps|oauth2.token_uri')
         client_id = Configuration.get('/ovs/framework/webapps|oauth2.client_id')
         client_secret = Configuration.get('/ovs/framework/webapps|oauth2.client_secret')
+        redirect_uri = 'https://{0}/api/oauth2/redirect/'.format(request.META['HTTP_HOST'])
         parameters = {'grant_type': 'authorization_code',
-                      'redirect_url': 'https://{0}/api/oauth2/redirect/'.format(System.get_my_storagerouter().ip),
-                      'client_id': client_id,
```

Comment:
REQUIRED, if the client is not authenticating with the
         authorization server as described in Section 3.2.1.

If you jump to seciont 3.2.1 and from there to 2.3 you will see its not required if you are passing the client_secret via authorization header.